File: CS330/6_tiling/Circle.java
package Pictures;

import java.awt.Color;
import java.awt.Graphics;

class Circle extends Shape
{
    public Circle(Color c)
    {
	super(c);
	_center = new java.awt.Point(0, 0);
	_radius = 0;
    }
       
   public Circle()
    {
	super(Color.black);
	_center = new java.awt.Point(0, 0);
	_radius = 0;
    }

   public Circle(java.awt.Point center, int radius)
    {
	super(Color.black);
	_center = (java.awt.Point)center.clone();
	_radius = radius;
    }

   public Circle(java.awt.Point center, int radius, Color c)
    {
	super(c);
	_center = (java.awt.Point)center.clone();
	_radius = radius;
    }


   public void move(int x, int y)
    {
	_center.x += x;
	_center.y += y;
    }

   public void plot(Graphics g)
    {
	g.setColor(getColor());
	g.drawOval(_center.x-_radius, _center.y-_radius,
		   2*_radius, 2*_radius);
    }

   public void reflectHorizontally (int xReflectionPlane)
    {
	_center = Shape.reflectHorizontally(_center, xReflectionPlane);
    }

   public void reflectVertically (int yReflectionPlane)
    {
	_center = Shape.reflectVertically(_center, yReflectionPlane);
    }

   public java.awt.Rectangle boundingBox()
    {
	return new java.awt.Rectangle(_center.x - _radius,
				      _center.y - _radius,
				      2*_radius, 2*_radius);
    }

   public String toString()
    {
	return "Circle: " + new Point(_center) + " " + _radius;
    }

   public Object clone() 
    {
	return new Circle(_center, _radius, getColor());
    }

private java.awt.Point _center;
private int   _radius;
};

File: CS330/_Old/3/circle.cpp
#include "circle.h"

#include <algorithm>

using namespace std;

Circle::Circle(Color edgeColor, Color fillColor)
 : Shape(edgeColor, fillColor)
{}


Circle::Circle(Point center, double radius, Color edgeColor, Color fillColor)
  : Shape(edgeColor, fillColor), _center(center), _radius(radius)
{}


void Circle::draw(Graphics& g) const
{
  if (getEdgeColor() != Color::transparent)
    {
      g.setColor(getEdgeColor());
      g.drawOval(round(_center.x()-_radius), round(_center.y()-_radius),
		 round(2*_radius), round(2*_radius));
    }
}

void Circle::fill(Graphics& g) const
{
  if (getFillColor() != Color::transparent)
    {
      g.setColor(getFillColor());
      g.fillOval(round(_center.x()-_radius), round(_center.y()-_radius),
		 round(2*_radius), round(2*_radius));
    }
}



void Circle::print(ostream& os) const
{
  os << "Circle: " << _center << " " << _radius;
}



void Circle::scale(const Point& center, double s)
{
  _center = _center.scale(center, s);
  _radius = (s*_radius);
}



void Circle::translate(double x, double y)
{
  _center = _center.translate(x, y);
}



RectangularArea Circle::boundingBox() const
{
  return RectangularArea(Point(_center.x()-_radius,
			       _center.y()-_radius),
			 Point(_center.x()+_radius,
			       _center.y()+_radius));
}

/*.............................................................*/

void Circle::get(std::istream& in)
{
  in >> _edge >> _fill;   // read edge and fill colors
  in >> _center >> _radius;
}